WordPress/WordPressTest/ContextManagerMock+AutomaticTeardown.swift

```diff
@@ -0,0 +1,72 @@
+import XCTest
+
+@testable import WordPress
+
+extension ContextManagerMock {
+    /// Override `ContextManager.shared` with the mock instance, and un-override it when
+    /// given `testCase` finishes.
+    ///
+    /// This method should be called during test set up, typically from `XCTestCase` subclass's
+    /// `setUp` method.
+    ///
+    /// - Parameter testCase: The test case to wait for.
+    @objc func useAsSharedInstance(untilTestFinished testCase: XCTestCase) {
+        // Create the test observer singleton to add it to `XCTestObservationCenter`.
+        _ = AutomaticTeardownTestObserver.instance
+
+        let original = ContextManager.overrideInstance
+        ContextManager.internalSharedInstance()
```

Comment:
Why do we call `internalSharedInstance()` here?

Given [the code of this method](https://github.com/wordpress-mobile/WordPress-iOS/blob/a6696d8b5ab8480ab0d1e93585b97a5a080e030d/WordPress/Classes/Utility/ContextManager.m#L68-L76), the only thing that it does would be to create a real sharedInstance inside the `dispatch_once`… instance that we'll never use in the test since we will override it in the next line in this code and further calls will return the `_override` object anyway.

I think we should be able to:
1. Remove the call to `ContextManager.internalSharedInstance()` from this `useAsSharedInstance(untilTestFinished:)` implementation
2. Take the occasion to optimize the original `internalSharedInstance` method to not create a real shared instance if `_override` is not `nil` because otherwise the creation of the never-used real shared instance seems quite useless and a waste:

```objc
// In ContextManager.m
+ (instancetype)internalSharedInstance
{
    if (_override) { return _override; }
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        _instance = [[ContextManager alloc] init];
    });
    return _instance;
}
```

Reply:
@mokagio and I also discussed why this `internalSharedInstance` call was necessary, see https://github.com/wordpress-mobile/WordPress-iOS/pull/18512#discussion_r865769366.

But since the problematic `dispatch_once` was removed in https://github.com/wordpress-mobile/WordPress-iOS/pull/18561, I don't think this call is needed anymore, I'll double check.

Regarding the `_override` in `ContextManager`, this static variable is actually not being used. The actual "override" happens [here at CoreDataHelper.swift](https://github.com/wordpress-mobile/WordPress-iOS/blob/a6696d8b5ab8480ab0d1e93585b97a5a080e030d/WordPress/Classes/Utility/CoreDataHelper.swift#L177).

Reply:
Ah. This warrants some cleanup then… or at least some more code comments to explain it, or to mark as `// FIXME` for later?

Reply:
Full test suite passed on my mac with this line commented out. I've removed it in [latest commit](https://github.com/wordpress-mobile/WordPress-iOS/pull/18579/commits/4d1f753a1e56d5026cdb3e35ef2a9a06fa80d637) 🥳

Reply:
The unused `_override` static variable [is also removed](https://github.com/wordpress-mobile/WordPress-iOS/pull/18579/commits/c6c1d4b01086a2e4c24d538e2a71a26e2c65a90e) to avoid confusion.